worksheet.rb

```diff
@@ -31,4 +42,134 @@
 # - the total amount of money each driver has made
 # - the average rating for each driver
 # - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# - Which driver has the highest average rating?
+
+driver_records = {
+    DR0001: [
+        {
+            date: "3rd Feb 2016",
+            cost: 10,
+            rider: "RD0003",
+            rating: 3
+        },
+        {
+            date: "3rd Feb 2016",
+            cost: 30,
+            rider: "RD0015",
+            rating: 4
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 45,
+            rider: "RD0003",
+            rating: 2
+        }
+    ],
+    DR0002: [
+        {
+            date: "3rd Feb 2016",
+            cost: 25,
+            rider: "RD0073",
+            rating: 5
+        },
+        {
+            date: "4th Feb 2016",
+            cost: 15,
+            rider: "RD0013",
+            rating: 1
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 35,
+            rider: "RD0066",
+            rating: 3
+        }
+    ],
+    DR0003: [
+        {
+            date: "4th Feb 2016",
+            cost: 5,
+            rider: "RD0066",
+            rating: 5
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 50,
+            rider: "RD0003",
+            rating: 2
+        }
+    ],
+    DR0004: [
+        {
+            date: "3rd Feb 2016",
+            cost: 5,
+            rider: "RD0022",
+            rating: 5
+        },
+        {
+            date: "4th Feb 2016",
+            cost: 10,
+            rider: "RD0022",
+            rating: 4
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 20,
+            rider: "RD0073",
+            rating: 5
+        }
+    ]
+}
+
+# - the number of rides each driver has given
```

Comment:
It looks like you pass the following three functions `driver_records`, and thus the parameter should have a name that reflects this.

Reply:
On that note, consider combining the following three methods into one to reduce repeated code. You could call it something like `calculate_individual_driver_stats`. It's a delicate balance between deciding to write many methods so that they each only do one thing and writing fewer methods to reduce repeated code.